src/_langs/en/updates/2015-07-22-measuring-performance-in-a-service-worker.markdown

```diff
@@ -0,0 +1,162 @@
+an overview.
+
+[https://www.youtube.com/watch?v=d5\_6yHixpsQ](https://www.youtube.com/watch?v=d5_6yHixpsQ)
+
+If you are going to Supercharge your Page load time as Jake suggests, you really 
+need to be able to understand how Service Workers affect your page's requests.
+
+The [Resource Timing API](http://w3c.github.io/resource-timing/#widl-PerformanceResourceTiming-workerStart)
+is a critcial component in many sites RUM (Real User Monitoring)
+infrastructure because it lets you understand holistically how all of your users see
+the performance of your site ([Another use-case is detecting content injection](https://paul.kinlan.me/detecting-injected-content/)).  In short, it lets you 
+understand nearly every aspect of every web request made from your site, well, unless you have a Service Worker.
+
+Here is a quick example of how it can be used in to get a list of all requests that were made
+to a domain that is not the current domain.
+
+{% highlight Javascript %}
+var getThirdPartyRequests = function() {
+  var thirdPartyRequests = [];
+  var requests = window.performance.getEntriesByType("resource");
+  
+  var currentHost = window.location.host
+
+  for(var requestIdx = 0; requestIdx < requests.length; requestIdx++) {
+    var request = requests[requestIdx];
+    var url = new URL(request.name);
+    var host = url.host;
+
+    if(host != currentHost) {
+      thirdPartyRequests.push(request);
+    }
+  }
+  
+  return thirdPartyRequests;
+};
+{% endhighlight %}
+ 
+The Resource Timing API was created and implemented before Service Worker was a twinkle in
+an engineers eye and the above code would not be able to understand how the Service Woker
+impacted it.  This is where a recent set of changes in Chrome 45 (Beta in July 2015) will help 
+you by introducing the ability for all forms of Workers (Web and Service Worker) 
+to have access to the Resource Timing API and thus be able to let you keep on top of 
+the network performance for all your users.
+
+Note: There are futher changes coming down the pipe in later versions of Chrome so 
+this article will be kept up to date.
+
+## Accessing Performance Metrics from a Service Worker
+
+The biggest change is the addition of the `performance` object into a Workers context (Web 
+and ServiceWorkers) that will now let you understand the performance timings of all 
+the requests that are made in the context of the worker. If you can only see 
+what happens from the context of the current window you would miss critical 
+timing information from:
+
+* `fetch()` requests made inside the `oninstall` event of the service worker
+* `fetch()` requests made when caching data in an `onpush` event can now be 
+  traced to understand the performance your users see.
+* finally, `fetch()` request that are made and intercepted by the `onfetch` 
+  handler.
+
+The last point is important. One way to think of a Service Worker is as a proxy 
```

Comment:
I'd love to this this paragraph much higher in the article, as it explains why you need to add extra perf monitoring logic when using SW.